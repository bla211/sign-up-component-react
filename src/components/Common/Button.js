import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
    render(){
        const { title, data, isError, onError } = this.props;
        let { onclick } = this.props;
        
        if(isError){
            onclick = () => {
                onError(true);
            }
        }


        return(
            <button onClick={ () => onclick(data) }>{ title }</button>
        )
    }

}

Button.propTypes = {
    title: PropTypes.string,
    onclick: PropTypes.func,
    data: PropTypes.string,
    isError: PropTypes.bool,
    onError: PropTypes.func
  };

export default Button;