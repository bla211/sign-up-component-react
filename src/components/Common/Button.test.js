import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Button from './Button';

configure({ adapter: new Adapter() });

describe('Test Button component', () => {
    test('Button renders {title}', () => {
        const title = "Next";
        const wrapper = shallow(<Button title={title}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Button fires onError function on click if "isError === true"', () => {
        const isError = true;
        const onError = jest.fn(); 
        const wrapper = shallow(<Button isError={isError} onError={onError}/>);
        wrapper.find('button').simulate('click');
        expect(onError).toHaveBeenCalled();
    });

    test('Button fires onclick function and passes {data} if no error', () => {
        const isError = false;
        const onclick = jest.fn();
        const data = 'payload'; 
        const wrapper = shallow(<Button isError={isError} onclick={onclick} data={data}/>);
        wrapper.find('button').simulate('click');
        expect(onclick).toHaveBeenCalledWith(data);
    });
});