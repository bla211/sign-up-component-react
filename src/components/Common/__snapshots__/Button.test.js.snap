// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Button renders {title} 1`] = `
<button
  onClick={[Function]}
>
  Next
</button>
`;
